inst/tests/tests.Rraw

```diff
@@ -21837,3 +21837,30 @@ DT[, V1000 := 20:1]
 test(2343.1, forderv(DT, by=names(DT), sort=FALSE, retGrp=TRUE), forderv(DT, by=c("V1", "V1000"), sort=FALSE, retGrp=TRUE))
 x = c(rep(0, 7e5), 1e6)
 test(2343.2, forderv(list(x)), integer(0))
+
+# Keep key when new column added before existing key in j
+# Incorrect key can lead to incorrect join result #7364
+local({
+  DT = data.table(V1 = 1:2, key = "V1")
+  test(2344.00, key(DT[, .(V2 = c("b", "a"), V1)]), "V1")
+  test(2344.01, key(DT[, .(V2 = -V1, V1)]), "V1")
+
+  d1 = data.table(
+    V1 = c(1L, 0L, 1L),
+    V2 = c("a", "a", "b"),
+    key = "V2"
+  )
+
+  d2 = d1[, .(
+    V1,
+    label = c("one", "zero", "one"),
+    V2
+  )]
+
+  r = d2[
```

Comment:
I was unsure if I should add this slightly more complicated test on the join result, which initially surfaced the bug or only keep the simpler test on the key itself.

Reply:
as long as the test itself is not taking ages, we add it most of the time!